mira/dkg/construct.py

```diff
@@ -211,10 +348,112 @@ def main(add_xref_edges: bool):
                 _header = next(reader)
                 writer.writerows(reader)
 
-    http_counter = Counter(http_nodes)
-    with HTTP_FAILURES_PATH.open("w") as file:
-        for url, count in http_counter.most_common():
-            print(url, count, sep="\t", file=file)
+    unstandardized_nodes_counter = Counter(unstandardized_nodes)
+    _write_counter(UNSTANDARDIZED_NODES_PATH, unstandardized_nodes_counter, title="url")
+
+    unstandardized_edges_counter = Counter(unstandardized_edges)
+    _write_counter(UNSTANDARDIZED_EDGES_PATH, unstandardized_edges_counter, title="url")
+
+    _write_counter(
+        EDGE_OBJ_COUNTER_PATH,
+        edge_target_usage_counter,
+        unpack=True,
+        title=("predicate", "predicate_label", "object_prefix"),
+    )
+    _write_counter(
+        SUB_EDGE_COUNTER_PATH,
+        subject_edge_usage_counter,
+        unpack=True,
+        title=("subject_prefix", "predicate", "predicate_label"),
+    )
+    _write_counter(
+        SUB_EDGE_TARGET_COUNTER_PATH,
+        subject_edge_target_usage_counter,
+        unpack=True,
+        title=("subject_prefix", "predicate", "predicate_label", "object_prefix"),
+    )
+    _write_counter(
+        EDGE_COUNTER_PATH,
+        edge_usage_counter,
+        unpack=True,
+        title=("predicate", "predicate_label"),
+    )
+
+    if do_upload:
+        upload_neo4j_s3()
+
+        from .construct_rdf import _construct_rdf
+        _construct_rdf(upload=True)
+
+        from .construct_registry import _construct_registry
+        _construct_registry(config_path=METAREGISTRY_PATH, upload=True)
+
+
+def _write_counter(
+    path: Path,
+    counter: Counter,
+    title: Union[None, str, Sequence[str]] = None,
+    unpack: bool = False,
+) -> None:
+    with path.open("w") as file:
+        if title:
+            if unpack:
+                print(*title, "count", sep="\t", file=file)
+            else:
+                print(title, "count", sep="\t", file=file)
+        for key, count in counter.most_common():
+            if unpack:
+                print(*key, count, sep="\t", file=file)
+            else:
+                print(key, count, sep="\t", file=file)
+
+
+def upload_s3(
+    path: Path, *, bucket: str = "askem-mira", intelligent_tiering: bool = False, s3_client=None
+) -> None:
+    """Upload the nodes and edges to S3."""
+    if s3_client is None:
+        import boto3
+
+        s3_client = boto3.client("s3")
+
+    today = datetime.today().strftime("%Y-%m-%d")
+    # don't include a preceding or trailing slash
+    key = f"dkg/epi/build/{today}/"
+    config = {
+        # https://stackoverflow.com/questions/41904806/how-to-upload-a-file-to-s3-and-make-it-public-using-boto3
+        "ACL": "public-read",
+    }
+    if intelligent_tiering:
+        config["StorageClass"] = "INTELLIGENT_TIERING"
```

Comment:
The current files on S3 seem to be using standard storage - I think we can use intelligent tiering here by default.

Reply:
Done in 95029d6